test/Microsoft.AspNetCore.Razor.Tools.Test/ServerLifecycleTest.cs

```diff
@@ -116,82 +116,86 @@ public void ServerStartup_SuccessfullyAcquiredMutex()
-                Assert.Equal(0, ((CompletedServerResponse)response).ReturnCode);
+                    // Assert
+                    var response = await compileTask;
+                    Assert.Equal(ServerResponse.ResponseType.Completed, response.Type);
+                    Assert.Equal(0, ((CompletedServerResponse)response).ReturnCode);
 
-                await serverData.Verify(connections: 2, completed: 2);
+                    await serverData.Verify(connections: 2, completed: 2);
+                }
             }
         }
 
         /// <summary>
         /// Multiple clients should be able to send shutdown requests to the server.
         /// </summary>
-        // Skipping temporarily on non-windows. https://github.com/aspnet/Razor/issues/1991
-        [ConditionalFact]
-        [OSSkipCondition(OperatingSystems.Linux)]
-        [OSSkipCondition(OperatingSystems.MacOSX)]
+        [Fact]
         public async Task ServerRunning_MultipleShutdownRequests_HandlesSuccessfully()
         {
             // Arrange
-            var completionSource = new TaskCompletionSource<bool>();
-            var host = CreateCompilerHost(c => c.ExecuteFunc = (req, ct) =>
-            {
-                // We want this to keep running even after the shutdown is seen.
-                completionSource.Task.Wait();
-                return EmptyServerResponse;
-            });
-
-            using (var serverData = ServerUtilities.CreateServer(compilerHost: host))
+            using (var startedMre = new ManualResetEventSlim(initialState: false))
+            using (var finishedMre = new ManualResetEventSlim(initialState: false))
             {
-                var compileTask = ServerUtilities.Send(serverData.PipeName, EmptyServerRequest);
+                var host = CreateCompilerHost(c => c.ExecuteFunc = (req, ct) =>
+                {
+                    // We want this to keep running even after the shutdown is seen.
+                    startedMre.Set();
+                    finishedMre.Wait();
+                    return EmptyServerResponse;
+                });
 
-                // Act
-                for (var i = 0; i < 10; i++)
+                using (var serverData = ServerUtilities.CreateServer(compilerHost: host))
                 {
-                    // The compilation is now in progress, send the shutdown.
-                    var processId = await ServerUtilities.SendShutdown(serverData.PipeName);
-                    Assert.Equal(Process.GetCurrentProcess().Id, processId);
-                    Assert.False(compileTask.IsCompleted);
-                }
+                    var compileTask = ServerUtilities.Send(serverData.PipeName, EmptyServerRequest);
+                    startedMre.Wait();
 
-                // Now let the task complete.
-                completionSource.SetResult(true);
+                    // Act
+                    for (var i = 0; i < 10; i++)
```

Comment:
Why is this in a loop?

Reply:
This is a test for testing multiple shutdown requests.